openvino_xai/explainer/visualizer.py

```diff
@@ -157,7 +159,24 @@ def visualize(
                 saliency_map_np = self._apply_colormap(explanation, saliency_map_np)
 
         # Convert back to dict
-        return self._update_explanation_with_processed_sal_map(explanation, saliency_map_np, class_idx_to_return)
+        return self._update_explanation_with_processed_sal_map(explanation, saliency_map_np, indices_to_return)
+
+    @staticmethod
+    def _apply_metadata(metadata: Dict[Task, Any], saliency_map_np: np.ndarray, indices: List[int | str]):
+        # TODO (negvet): support when indices are strings
+        if metadata:
+            if Task.DETECTION in metadata:
+                for smap_i, target_index in zip(range(len(saliency_map_np)), indices):
+                    saliency_map = saliency_map_np[smap_i]
+                    box, score, label_index = metadata[Task.DETECTION][target_index]
+                    x1, y1, x2, y2 = box
+                    cv2.rectangle(saliency_map, (int(x1), int(y1)), (int(x2), int(y2)), color=(255, 0, 0), thickness=2)
+                    box_name = f"{label_index}|{score:.2f}"
+                    org = int(x1), int(y1 - 5)
+                    cv2.putText(
+                        saliency_map, box_name, org=org, fontFace=1, fontScale=1, color=(255, 0, 0), thickness=2
+                    )
+        return saliency_map_np
```

Comment:
Flattened the condition and renamed variables for clarity.
We can return just `saliency_map_np` because we update `saliency_map` that is a link to `saliency_map_np[sal_map_i]`?
For better clarity, can we add
`saliency_map_np[sal_map_i] = saliency_map` in the end of the loop?

```suggestion
        if metadata and Task.DETECTION in metadata:
              for sal_map_i, target_index in zip(range(len(saliency_map_np)), indices):
                  saliency_map = saliency_map_np[sal_map_i]
                  bbox, score, label_index = metadata[Task.DETECTION][target_index]
                  x1, y1, x2, y2 = bbox
                  cv2.rectangle(saliency_map, (int(x1), int(y1)), (int(x2), int(y2)), color=(255, 0, 0), thickness=2)
                  bbox_label = f"{label_index}|{score:.2f}"
                  bbox_label_loc = int(x1), int(y1 - 5)
                  cv2.putText(
                      saliency_map, bbox_label, org=bbox_label_loc, fontFace=1, fontScale=1, color=(255, 0, 0), thickness=2
                  )
                  saliency_map_np[sal_map_i] = saliency_map
        return saliency_map_np
```

Reply:
I also do not like extra indention level that I have here, but I kept it here to further extend for classification task. `saliency_map_np` is numpy array, which is a mutable object, that is why if we update its slices, it will be updated as well (hope it is clear). Thanks for the variable renaming, I will adopt.